### PR TITLE
ref: Disable Mono StackTrace Factory.

### PR DESCRIPTION
**Pull request by lucas-zimerman**
As discussed on Discord, the MonoStackTraceFactory lacks any Assembly information, generating a different grouping compared to the previous stable release (2.1.8).

In addition, since there's no assembly info, the SDK doesn't exclude the known frames that are not part of the user code.
To conclude, the current SentryStackTraceFactory produces richer information when running on mono compared to MonoStackTraceFactory, but it's only lacking the additional info for symbolicate the frames. But since this code wasn't implemented on the server-side, I feel like it's better, for now, to stick to the current StackTraceFactory for mono and in the future, once the symbolization code is ready, enhance the current StackTrace with the needed information.

### CODE REVIEW

**Comment by bruno-garcia** [comment]
> generating a different grouping compared to the previous stable release (2.1.8).

Shouldn't affect grouping though. Did you confirm it does?
It just doesn't show the assembly info UI.

**Comment by bruno-garcia** [comment]
As we agreed on Discord, please just don't enabled it on `IsMono`?

The code that's there will likely still be needed on IL2CPP or some other Mono-like environment where the `new StackTrace(.., true)` won't work as expected.

File: src/Sentry/SentryOptions.cs [hunk 1]
```diff
@@ -436,10 +436,7 @@ public SentryOptions()
                 () => ExceptionProcessors ?? Enumerable.Empty<ISentryEventExceptionProcessor>()
             };
 
-            SentryStackTraceFactory = Runtime.Current.IsMono()
-                // Also true for IL2CPP
-                ? new MonoSentryStackTraceFactory(this)
-                : new SentryStackTraceFactory(this);
+            SentryStackTraceFactory = new SentryStackTraceFactory(this);
```
Comment (bruno-garcia) [inline src/Sentry/SentryOptions.cs:439]: This is the only change we need actually. Just don't add it by default on Mono but keep the code around for now until we're sure we won't need this for Unity

**Review by bruno-garcia — CHANGES_REQUESTED** [review]
This is the only change we need actually. Just don't add it by default on Mono but keep the code around for now until we're sure we won't need this for Unity

**Comment by lucas-zimerman** [comment]
Fixxd

**Review by bruno-garcia — APPROVED** [review]
Thanks @lucas-zimerman

**Pull request merged**